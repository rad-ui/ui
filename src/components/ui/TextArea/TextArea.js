'use client';
import React from 'react';
const TextArea = ({children})=>{
    return <textarea>

    </textarea>;
};

export default TextArea;
