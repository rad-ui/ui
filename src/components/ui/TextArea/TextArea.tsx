'use client'

import React from 'react'

export type TextAreaProps = {
  children: React.ReactNode
}

const TextArea = ({ children }: TextAreaProps) => {
  return <textarea>{children}</textarea>
}

export default TextArea
