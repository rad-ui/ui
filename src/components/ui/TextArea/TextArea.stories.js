import React from 'react';
import TextArea from './TextArea';
import SandboxEditor from '~/components/tools/SandboxEditor/SandboxEditor';


const Template = (args) => {
    return <SandboxEditor className="space-y-4 pt-4">
        <TextArea {...args} >

        </TextArea>
    </SandboxEditor>;
};

export default {
    title: 'UI/Input/TextArea',
    component: TextArea,
    render: (args) => <Template {...args} />,
};

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const All = {
    args: {
        className: '',
        placeholder: 'Type something here',
    },
};
