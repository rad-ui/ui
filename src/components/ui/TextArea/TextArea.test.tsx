import React from 'react';
import { render, screen } from '@testing-library/react';
import TextArea from './TextArea';

describe('TextArea', () => {
    it('forwards ref to the root element', () => {
        const ref = React.createRef<HTMLDivElement>();
        render(<TextArea ref={ref}>content</TextArea>);
        expect(ref.current).toBeInstanceOf(HTMLDivElement);
    });

    it('forwards ref to the input element', () => {
        const ref = React.createRef<HTMLTextAreaElement>();
        render(<TextArea.Input ref={ref} placeholder="test" />);
        expect(ref.current).toBeInstanceOf(HTMLTextAreaElement);
    });

    it('forwards ref to the root subcomponent', () => {
        const ref = React.createRef<HTMLDivElement>();
        render(<TextArea.Root ref={ref}>child</TextArea.Root>);
        expect(ref.current).toBeInstanceOf(HTMLDivElement);
    });

    it('is accessible via placeholder', () => {
        render(<TextArea>hidden</TextArea>);
        expect(screen.getByPlaceholderText('enter text')).toBeInTheDocument();
    });

    it('renders without console warnings', () => {
        const warn = jest.spyOn(console, 'warn').mockImplementation(() => {});
        const error = jest.spyOn(console, 'error').mockImplementation(() => {});
        render(<TextArea>content</TextArea>);
        expect(warn).not.toHaveBeenCalled();
        expect(error).not.toHaveBeenCalled();
        warn.mockRestore();
        error.mockRestore();
    });

    it('matches snapshot', () => {
        const { container } = render(<TextArea>content</TextArea>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
