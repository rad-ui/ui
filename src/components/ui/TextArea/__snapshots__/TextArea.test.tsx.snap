// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea matches snapshot 1`] = `
<div
  class="rad-ui-text-area"
  data-textarea-resize="both"
>
  <textarea
    placeholder="enter text"
  >
    content
  </textarea>
  content
</div>
`;
